### PR TITLE
Add the remaining jdbc data-sources to kubernetes service bindings

### PR DESCRIPTION
**Pull request by geoand**
Follows up on https://github.com/quarkusio/quarkus/pull/13737

### CODE REVIEW

**Comment by geoand** [comment]
The problem I have with this work, is that all the jdbc drivers will have very similar code and hate copying and pasting everywhere... 
On the other hand  I am not sure I  want to introduce a jar to put the common code in...

@machi1990 @gsmet WDYT?

**Comment by geoand** [comment]
Bumping this :)

**Comment by machi1990** [comment]
LGTM.

> The problem I have with this work, is that all the jdbc drivers will have very similar code and hate copying and pasting everywhere...

I think the code will be pretty similar and it make sense to remove the duplication.

> On the other hand I am not sure I want to introduce a jar to put the common code in...
 What makes you not want to introduce a jar? It may be not worth it for two service bindings for jdbc drivers we have but we more drivers will need this capability and it maybe hard to maintain ;-)

> Bumping this :)

Thanks, only seeing this now. Was mostly away the past couple of weeks.

**Comment by geoand** [comment]
> I think the code will be pretty similar and it make sense to remove the duplication.

My problem is however that I don't know where such "utility" code would go...

> Thanks, only seeing this now. Was mostly away the past couple of weeks.

👍🏼

**Comment by geoand** [comment]
@gsmet any ideas?

**Comment by geoand** [comment]
This is the issue we talked about @yharish991

**Comment by geoand** [comment]
Folks, can I have some feedback on this?

Otherwise I will just have to copy and paste similar code to all the datasources

**Comment by machi1990** [comment]
> Folks, can I have some feedback on this?

I'll consider the SPI / common jar approach but @gsmet might have better ideas. 
 
> Otherwise I will just have to copy and paste similar code to all the datasources

I am fine with it for now if we do not have a better idea to/than the common jar.

**Comment by FroMage** [comment]
@geoand just put it in quarkus-kubernetes-service-binding since it doesn't depend on anything besides that module.

**Comment by geoand** [comment]
> @geoand just put it in quarkus-kubernetes-service-binding since it doesn't depend on anything besides that module.

I'll check it out. I am pretty sure there was a reason I didn't do that, but I can't remember what it was.

**Comment by geoand** [comment]
> > @geoand just put it in quarkus-kubernetes-service-binding since it doesn't depend on anything besides that module.
> 
> I'll check it out. I am pretty sure there was a reason I didn't do that, but I can't remember what it was.

Indeed this is what I'll do - I'll just add a Datasource converter util in `quarkus-kubernetes-service-binding`

**Comment by geoand** [comment]
PR updated to include all the relevant jdbc datasources

File: extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/JdbcDatasourceUtil.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        String host = binding.getProperties().get("host");
+        String port = binding.getProperties().get("port");
+        String database = binding.getProperties().get("database");
+        if ((host != null) && (database != null)) {
+            String portPart = "";
+            if (port != null) {
+                portPart = ":" + port;
+            }
+            properties.put("quarkus.datasource.jdbc.url",
+                    String.format("jdbc:%s://%s%s/%s", urlType, host, portPart, database));
```
Comment (gsmet) [inline extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/JdbcDatasourceUtil.java:56]: Does this really work with all the datasources? I'm wondering if the datasource shouldn't be providing the format?

Comment (geoand) [reply extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/JdbcDatasourceUtil.java:56]: It seems to be common for all the ones we support, no?

Comment (gsmet) [reply extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/JdbcDatasourceUtil.java:56]: Maybe. If you have checked them then we can probably live with it for the time being.

Comment (geoand) [reply extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/JdbcDatasourceUtil.java:56]: I checked the URLs we use in the integration tests and they are all the same.

I agree that long term we'll probably need a better strategy (which most likely will that into the various DB specifc properties that Service Binding might give us), but for the time being I think it's OK

**Comment by geoand** [comment]
Can I have this in for `1.11.1` please?

**Comment by geoand** [comment]
> Can I have this in for `1.11.1` please?

Can we at least have it `master` and potentially in `1.11.2`?

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Sure, next time, add the `backport?` label so that it's on my radar: I was focused on converging on the backportable PRs.

**Comment by geoand** [comment]
Great, thanks

**Pull request merged**